scvi/dataset/dataset.py

```diff
@@ -1090,6 +1117,21 @@ def map_cell_types(
         for cell_types, new_cell_type_name in cell_types_dict.items():
             self.merge_cell_types(cell_types, new_cell_type_name)
 
+    def reorder_cell_types(self, new_order: Union[List[str], np.ndarray]):
+        """Reorder in place the cell-types. The cell-types provided will be added at the beginning of `cell_types`
+        attribute, such that if some existing cell-types are omitted in `new_order`, they will be left after the
+        new given order
+
+        :param new_order:
+        """
+        for cell_type in self.cell_types:
+            if cell_type not in new_order:
+                new_order.append(cell_type)
```

Comment:
you typed `new_order` as a potential `np.ndarray`. You need to type check before calling `append` and treat the `numpy` case accordingly

Reply:
Good catch, thank you!  `numpy` arrays are not designed to be resized and for dynamic changes so the best is to convert it in a list at the beginning. Thanks!